bundle/edu.gemini.ags.client.impl/src/main/scala/edu/gemini/ags/client/impl/AgsHttpClient.scala

```diff
@@ -39,12 +47,18 @@ case class AgsHttpClient(host: String, port: Int) extends AgsClient {
   private def estimateNow(url: URL): AgsResult = {
     try {
       AgsHttpClient.Log.info(s"AGS Query to $url")
-      val conn = url.openConnection().asInstanceOf[HttpsURLConnection]
-      conn.setHostnameVerifier(hostnameVerifier)
-      conn.setSSLSocketFactory(GemSslSocketFactory.get)
-      conn.setReadTimeout(timeout)
-      Charset.set(conn)
-      Response(conn).result
+      Option(lRUCache.get(url.toString) : AgsResult) | {
+        val conn = url.openConnection().asInstanceOf[HttpsURLConnection]
+        conn.setHostnameVerifier(hostnameVerifier)
+        conn.setSSLSocketFactory(GemSslSocketFactory.get)
+        conn.setReadTimeout(timeout)
+        Charset.set(conn)
+
+        Response(conn).result match {
+          case s@AgsResult.Success(_) => lRUCache.put(url.toString, s)
```

Comment:
I think overkill given that the computation to get the result is doing network access anyway and that isn't done in IO.  Perhaps if I'd known better I would have written this code differently in the first place but as is, I don't think there is anything to be gained by wrapping that bit in IO.